doc/whats_new/v1.1.rst

```diff
@@ -756,6 +756,11 @@ Changelog
   `check_methods_sample_order_invariance` and `check_methods_subset_invariance` tests.
   :pr:`22318` by `Zhehao Liu <MaxwellLZH>`.
 
+- |Fix| Changes the error message of the ValidationError raised by
+  :func:`utils.validation.check_X_y` when y is None so that it is compatible
+  with :func:`utils.estimator_checks.check_requires_y_none`. :pr:`22578` by
+  :user:`Claudio Salvatore Arcidiacono <ClaudioSalvatoreArcidiacono>`.
```

Comment:
```suggestion
- |Fix| Changes the error message of the `ValidationError` raised by
  :func:`utils.validation.check_X_y` when y is None so that it is compatible
  with :func:`utils.estimator_checks.check_requires_y_none`. :pr:`22578` by
  :user:`Claudio Salvatore Arcidiacono <ClaudioSalvatoreArcidiacono>`.
```